src/monodroid/jni/android-system.h

```diff
@@ -119,18 +131,52 @@ namespace xamarin { namespace android { namespace internal
 		int setenv (const char *name, const char *value, int overwrite);
 #endif
 
+		bool is_mono_llvm_enabled () const
+		{
+			return knownEnvVars.MonoLLVM;
+		}
+
+		bool is_embedded_dso_mode_enabled () const
+		{
+			return knownEnvVars.DSOInApk;
+		}
+
+		MonoAotMode get_mono_aot_mode () const
+		{
+			if (knownEnvVars.MonoAOT == nullptr || *knownEnvVars.MonoAOT == '\0')
```

Comment:
Instead of `strdup()`ing a string into `knownEnvVars.MonoAOT`, should the string instead be mapped into a `MonoAotMode` enum so that we don't need to re-`switch` on the first character of `knownEnvVars.MonoAOT`?  It would save memory, at least.